src/1password.ts

```diff
@@ -0,0 +1,178 @@
+import * as core from '@actions/core'
+// import * as exec from '@actions/exec'
+import {install} from './install'
+import * as tc from '@actions/tool-cache'
+import {execWithOutput} from './exec'
+import {Item} from './types'
+import * as parsing from './parsing'
+
+const ONE_PASSWORD_VERSION = '1.8.0'
+
+export class OnePassword {
+  onePasswordEnv: {[key: string]: string}
+
+  constructor(deviceId: string) {
+    this.onePasswordEnv = {
+      OP_DEVICE: deviceId
+    }
+
+    if (process.env['XDG_CONFIG_HOME'] === undefined) {
+      // This env var isn't set on GitHub-hosted runners
+      this.onePasswordEnv.XDG_CONFIG_HOME = `${process.env['HOME']}/.config`
+    }
+  }
+
+  async setupAndInstallIfNeeded(): Promise<void> {
+    // Check if op is installed and download if necessary
+    const cachedOpDirectory = tc.find('op', ONE_PASSWORD_VERSION)
+    // This seems like a weird API, why not return undefined?
+    if (cachedOpDirectory !== '') {
+      core.addPath(cachedOpDirectory)
+    } else {
+      await install(ONE_PASSWORD_VERSION)
+    }
+  }
+
+  async signIn(
+    signInAddress: string,
+    emailAddress: string,
+    secretKey: string,
+    masterPassword: string
+  ): Promise<void> {
+    const env = this.onePasswordEnv
+    try {
+      const output = await execWithOutput(
+        'op',
+        [
+          'signin',
+          signInAddress,
+          emailAddress,
+          secretKey,
+          '--raw',
+          '--shorthand',
+          'github_action'
+        ],
+        {
+          env,
+          input: Buffer.alloc(masterPassword.length, masterPassword)
+        }
+      )
+      core.info('Successfully logged into 1Password')
+      const session = output.toString().trim()
+      core.setSecret(session)
+
+      this.onePasswordEnv.OP_SESSION_github_action = session
+    } catch (error) {
+      throw new Error(error)
+    }
+  }
+
+  async requestItems(itemRequests: parsing.ItemRequest[]): Promise<void> {
+    for (const itemRequest of itemRequests) {
+      try {
+        const itemsJSON = await this.getVault(itemRequest.vault)
+
+        const items: Item[] = JSON.parse(itemsJSON)
+        const uuid = items
+          .filter(item => item.overview.title === itemRequest.name)
+          .map(item => item.uuid)[0]
+
+        if (!uuid) {
+          throw new Error(
+            `Could not find item in vault \u001b[38;2;255;255;0m'${itemRequest.vault}' \u001b[38;2;255;255;255mwith name \u001b[38;2;255;255;0m'${itemRequest.name}'`
+          )
+        }
+
+        core.info(`Loading ${itemRequest.name} detail`)
```

Comment:
Could this be moved into `getItemInVault` for consistency with `getVault`?

Reply:
slight change, I moved all the `core.info` outside of the specific calls. Otherwise, I had to pass in more properties to print what I wanted to.